pkgs/by-name/ma/material-maker/package.nix

```
@@ -0,0 +1,138 @@
+  libXfixes,
+  libXi,
+  libXinerama,
+  libXrandr,
+  libXrender,
+  alsa-lib,
+  libpulseaudio,
+  udev,
+}:
+let
+  pname = "material-maker";
+  version = "1.3";
+  versionUnderscore = "${lib.replaceStrings [ "." ] [ "_" ] version}";
+
+  sources = {
+    linux = fetchzip {
+      url = "https://github.com/RodZill4/material-maker/releases/download/${version}/material_maker_${versionUnderscore}_linux.tar.gz";
+      hash = "sha256-WEu5gVfnswB5zYzu3leOL+hKOBzJbn48gHQKshlfOh4=";
+    };
+    darwin = fetchurl {
+      url = "https://github.com/RodZill4/material-maker/releases/download/${version}/material_maker_${versionUnderscore}.dmg";
+      hash = "sha256-D4jPQEOKws2JFczh0K55mR08jIBMU6jOFOm6TyFxnt8=";
+    };
+  };
+in
+stdenv.mkDerivation (finalAttrs: {
+  inherit pname version;
+
+  src =
+    if stdenv.isLinux && stdenv.isx86_64 then
+      sources.linux
+    else if stdenv.isDarwin then
+      sources.darwin
+    else
+      throw "unsupported platform";
+
+  nativeBuildInputs = [
+    autoPatchelfHook
+    copyDesktopItems
+  ];
+
+  buildInputs = [
+    libGL
+    libGLU
+    libX11
+    libXcursor
+    libXext
+    libXfixes
+    libXi
+    libXinerama
+    libXrandr
+    libXrender
+  ];
+
+  desktopItems = makeDesktopItem {
+    name = "material-maker";
+    exec = "material-maker";
+    icon = "material-maker";
+    genericName = "Procedural texture generation and 3D model painting tool";
+    desktopName = "Material Maker";
+    comment = finalAttrs.meta.description;
+    terminal = false;
+    categories = [
+      "Graphics"
+      "3DGraphics"
+    ];
+    keywords = [ "material" ];
+  };
+
+  unpackPhase =
+    if stdenv.isDarwin then
+      ''
+        ${pkgs.undmg}/bin/undmg $src
+      ''
+    else
+      "";
+
+  installPhase =
+    ''
+      runHook preInstall
+    ''
+    + lib.optionalString stdenv.isLinux ''
+      install -D -m 755 $src/material_maker.x86_64 $out/libexec/material-maker
+      install -D -m 644 $src/material_maker.pck $out/libexec/material-maker.pck
+
+      for dir in examples library nodes doc environments meshes export; do
+        cp -R $src/$dir $out/libexec
+      done
+
+      install -D -m 644 $src/doc/_static/icon.png -t $out/share/icons/hicolor/256x256/apps/
+
+      mkdir -p $out/bin
+      ln -s $out/libexec/material-maker $out/bin/material-maker
+    ''
+    + lib.optionalString stdenv.isDarwin ''
+      mkdir -p $out/Applications
+      cp -r ./material_maker.app $out/Applications
+    ''
+    + ''
+      runHook postInstall
+    '';
+
+  runtimeDependencies = map lib.getLib [
+    alsa-lib
+    libpulseaudio
+    udev
+  ];
+
+  passthru.updateScript = nix-update-script { };
+
+  meta = {
+    homepage = "https://www.materialmaker.org/";
+    description = "Tool based on Godot Engine that can be used to create textures procedurally and paint 3D models";
+    license = lib.licenses.mit;
+    platforms = lib.platforms.darwin ++ (lib.intersectLists lib.platforms.linux lib.platforms.x86_64);
```

```suggestion
    platforms = lib.platforms.darwin ++ (lib.intersectLists lib.platforms.linux lib.platforms.x86_64);
    sourceProvenance = with lib.sourceTypes; [ binaryNativeCode ];
```

See https://nixos.org/manual/nixpkgs/stable/#sec-meta-sourceProvenance